### PR TITLE
Allow sonarqube addon on external PRs

### PR DESCRIPTION
**Pull request by BanzaiMan**
In this case, addon runs without sensitive data

### CODE REVIEW

**Comment by BanzaiMan** [comment]
This PR replaces https://github.com/travis-ci/travis-scheduler/pull/42.

**Comment by svenfuchs** [comment]
Looks GUT! ❤️

**Comment by gortiz** [comment]
Hi there,

Sorry to revive this thread but... is this feature released on Travis? The documentation still says that SonarQube addon is not working on external PRs. The last reference I can find about that are, in fact, latter than this PR (https://github.com/travis-ci/docs-travis-ci-com/pull/825)

**Comment by BanzaiMan** [comment]
@gortiz This is live.

The documentation, from the PR, reads:

> For security reasons, this advanced feature works only for INTERNAL pull requests. In other words, pull requests built from forks won’t be inspected.

The addon runs on external PRs, but without sensitive data.

Did you try it, or are you just asking?

**Comment by gortiz** [comment]
Hi @BanzaiMan 

I am a developer of ToroDB (github.com/torodb/torodb) and I would like to review PRs so we can slowly decrease our technical debt, but on this repository is usual that even developers with admin privileges create PRs from their own forks.

I didn't try to add the addon (in fact it is added, but only on devel and master branch), but reading the documentation I understand that the external PRs are not going to be analyzed. What do you mean when saying that the addon run but without sensitive data? Did I misunderstood the documentation?

**Comment by BanzaiMan** [comment]
@bellingard @dbmeneses Could you elaborate on the limitation on SonarQube on external PRs? Thanks!

**Comment by bellingard** [comment]
@gortiz To be able to analyse a PR, the SonarQube add-on currently needs 2 things: one GitHub token (to comment the PR) and one SonarQube.com token (to get settings for you project). These is the sensitive data @BanzaiMan is referring to. 

Because no one can predict what a PR built from a fork will do (like adding `echo $SONAR_TOKEN` in your `.travis.yml` file), this sensitive data is not available in this context, and therefore the SonarQube add-on just makes sure that if a `sonar-scanner` or `mvn sonar:sonar` execution is triggered, the analysis gently fails silently (purpose is to not fail your PR builds for useless reasons).

So to sum up:
* If you have PR directly issued from branches of your repository, then they will be analysed successfully because the sensitive data is available (code is trusted)
* For any other PR, analysis will be ignored (= fails silently)

We'll be working soon on that topic to enable PR analysis wherever the PR comes from.

**Comment by gortiz** [comment]
Hi @bellingard,

Ok, I understand, it is as I expected. When I read this issue summary, I thought that it was my lucky day and SonarQube addon was already working on external PR but it was just undocumented. I guess I wasn't that lucky, but I will be looking forward the version that include this feature.

Thanks
